File: 28-FindtheIndexoftheFirstOccurrenceinaString/28-FindtheIndexoftheFirstOccurrenceinaString.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        for (int i = 0; i < haystack.length(); i++){
            if (haystack.length() - i < needle.length()){
                return -1;
            }
            if (haystack[i] == needle[0]){
                for (int j = 0; j < needle.length(); j++){
                    if (haystack[i+j]==needle[j] && j == needle.length() -1){
                        return i;
                    }
                    if (haystack[i+j]==needle[j]){
                        continue;
                    }
                    else{
                        break;
                    }
                }


            }
        }
        return -1;
    }
};

File: 28-FindTheIndexOfTheFirstOccurrenceInAString/28-FindTheIndexOfTheFirstOccurrenceInAString.py
# Last updated: 7/2/2026, 11:17:56 PM
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        #needle's entire legnth has to be able to fit within haystack
        for i in range(len(haystack) - len(needle) + 1):
            #for each iteration we only look at the length of the needle starting at that round of iteration
            if haystack[i:i+len(needle)] == needle:
                return i

        return -1